ext/pdo_firebird/firebird_statement.c

```diff
@@ -459,11 +464,11 @@ static int firebird_stmt_get_col(
 					break;
 				case SQL_FLOAT:
 					/* TODO: Why is this not returned as the native type? */
-					ZVAL_STR(result, zend_strpprintf(0, "%F", *(float*)var->sqldata));
+					ZVAL_STR(result, zend_strpprintf_unchecked(0, "%.*H", get_float_from_sqldata(var->sqldata), 8));
```

Comment:
I believe you can use "%.8H" instead of "%.*H" (and get rid of the argument 8).

Reply:
Yes, but in that case, it seemed to use a macro that converted a string to a number, so I thought it would be more efficient to pass a number as an argument from the beginning.

TBH, I don't know how much the efficiency will change. Personally, I think either way of writing is fine, but do you think there is much difference in terms of efficiency?

Reply:
That's fair, then we can leave this as-is. I don't think the efficiency between the two differs noticeably though.

Reply:
I don't really understand, but after correcting this, the test that was failing on Windows_ZTS passed.